src/SystemManager.hh

```diff
@@ -41,6 +43,21 @@ namespace gz
     // Inline bracket to help doxygen filtering.
     inline namespace GZ_SIM_VERSION_NAMESPACE {
 
+    /// \brief Helper container to keep track of
+    /// system interfaces and their parents
+    template<typename IFace>
+    struct SystemHolder {
```

Comment:
`SystemHolder` isn't the clearest name; it sounds very much like `SystemContainer`, but it doesn't hold systems. I think `SystemIfaceWithParent` is more accurate, though not super concise

Reply:
Not sure that adding SystemHolder is neccessary. All the info about system->parentId mapping is already stored in the SystemManager::systems container. 

So in SystemManager::ProcessRemovedEntities(), what about just filtering SystemManager::systems and then clearing+rebuilding the interface containers (systemsConfigure, systemsReset) etc from scratch, like in SystemManager::ActivatePendingSystems() ?

Reply:
you are correct that the `SystemInternal` data structure held by the `systems` member variable does include the parent entity, but this `SystemHolder` data structure doesn't store `SystemInternal` data. My first comment suggests renaming this data structure, since it doesn't hold a `gz::sim::System` or `SystemInternal`; rather it stores an interface pointer and `Entity` Id

Reply:
and to add why this is needed; it's currently augmenting the data structures that hold a vector of interface pointers

Reply:
I think @landersson suggestion may have some merit. Unfortunately I discovered a bug in my `SystemContainer` implementation which is taking me a bit of time to debug  while implementing it.

Reply:
> I think @landersson 's suggestion may have some merit (thanks @landersson).

👍 

> Unfortunately I discovered a bug in my `SystemContainer` implementation which is taking me a bit of time to debug while implementing it.
> 
> Essentially since we dont run remove entities very often it should be pretty ok to flush and rebuild the `SystemIface` list. Either ways, I will debug `SystemContainer` first (Fixed in [4b4332d](https://github.com/gazebosim/gz-sim/pull/2232/commits/4b4332da7785985283a26ca79614c9fc370bbf7d)).

ok sounds good

Reply:
Unfortunately I think its been too long since I touched this PR. The runner does need access to the parent information hence we need the `SystemIfaceWithParent` class. Although it'd have been nice not to have it.